03_inheritance/03_03_freeform_inheritance.py

```diff
@@ -12,3 +13,55 @@
 #
 # - A `Vehicle()` parent class, with `Truck()` and `Motorcycle()` child classes.
 # - A `Restaurant()` parent class, with `Gourmet()` and `FastFood()` child classes.
+
+class Restaurant:
+    """Restaurants"""
+    def __init__(self, name, region, to_go, price, meal) -> None:
+        self.name = name
+        self.region = region # Part of the world where food is from
+        self.to_go = to_go # Y or N
+        self.price = price # 1-$, 2-$$, 3-$$$
+        self.meal = meal # breakfst, lunch or dinner
+
+    def __str__(self) -> str:
+        return f"{self.name} {self.region} {self.drive_thru} {self.to_go} {self.price} {self.meal}"
+
+class FastFood(Restaurant):
+    def __init__(self, name, region, to_go, meal, price=1, drive_thru=True) -> None:
+        super().__init__(name, region, to_go, price, meal)
+        self.drive_thru = drive_thru
+
+class BurgerJoint(FastFood):
+    def __init__(self, name, to_go, meal, burger_meat, region = "American", price=1, drive_thru=True) -> None:
+        super().__init__(name, region, to_go, meal, price, drive_thru)
+        self.burger_meat = burger_meat
+
+class Gourmet(Restaurant):
+    def __init__(self, name, region, to_go, meal, valet, price=3) -> None:
+        super().__init__(name, region, to_go, price, meal)
+        self.valet = valet
+
+
+mcdonalds = FastFood(
+    name= "McDonalds", 
+    region= "American", 
+    to_go=True,
+    meal= ["breakfast", "lunch", "dinner"])
+
+in_n_out = BurgerJoint(
+    name= "In N Out",
+    to_go= True,
+    meal = ["lunch", "dinner"],
+    burger_meat = ["beef", "vegetarian", "buffalo"])
+
+la_loma = Gourmet(
+    name = "La Loma",
+    region= "Mexican",
+    to_go= True,
+    meal = ["breakfast", "lunch", "dinner"],
+    valet= True)
```

Comment:
Well done for using keyword arguments here!

**nitpick**: It's nice to stay consistent with formatting.

```suggestion
mcdonalds = FastFood(
    name = "McDonalds", 
    region = "American", 
    to_go =True,
    meal = ["breakfast", "lunch", "dinner"])

in_n_out = BurgerJoint(
    name = "In N Out",
    to_go = True,
    meal = ["lunch", "dinner"],
    burger_meat = ["beef", "vegetarian", "buffalo"])

la_loma = Gourmet(
    name = "La Loma",
    region = "Mexican",
    to_go = True,
    meal = ["breakfast", "lunch", "dinner"],
    valet = True)
```

Reply:
Ok, silly question. How do I make the "adjustments"? Am I accepting your "commit suggestion"'s or am I editing it myself (and where would I do that?)

Reply:
You can either "commit suggestion" which will create a commit with my change. If you do this, you need to remember to pull from your local version.

Or you can edit it yourself by:

- Going to your local repo
- Switching to the branch that this PR was made from (`inheretence_branch`)
- Making the change
- Making a commit and pushing

This PR will then update with that commit automatically.